runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -669,8 +702,32 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 		}
 	}
 
-	private void createGameTimer(GameTimer timer)
+	/**
+	 * Adds a game timer to the infobox manager to render. Accepts a `replaceExisting` option to
+	 * preserve existing timers of the same type instead of replacing them with the new passed
+	 * timer.
+	 *
+	 * @param timer           			The timer to be added to the infobox manager.
+	 * @param replaceExisting 			Whether to remove existing timer of the same type. `true` to replace
+	 *                        			it, `false` to preserve it if present.
+	 * @param checkActiveFreezeTimers	Whether to check for active freeze timers before creating a new one,
+	 *                                  potentially of a different spell type (i.e.: blitz vs. barrage)
+	 */
+
+	private void createGameTimer(final GameTimer timer, final boolean replaceExisting, final boolean checkActiveFreezeTimers)
```

Comment:
Since this isn't a general createGameTimer method (can't be applied to other timer types), this might be better-suited as a differently-named method, and the `removeExisting` code moved back into `createGameTimer(GameTimer, boolean)`.

Reply:
Agreed. 

You could also argue that the signature `replaceExisting` is also only used for freeze timers as well. However, in the future we may want similar behavior for non-freezing timers and nothing in that function specifically checks for freezes, so doesn't need to be changed. Thoughts?

Reply:
In that case, I'd want to see this become something more generic, like accepting a set of GameTimers that should be checked against.

Reply:
Ok, about to push some code up!